src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -0,0 +1,412 @@
+package mesosphere.mesos
+
+import mesosphere.marathon.MarathonConf
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.pod.PodDefinition
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.raml
+import mesosphere.marathon.state.{ EnvVarString, PathId, Timestamp }
+import mesosphere.mesos.ResourceMatcher.ResourceSelector
+import org.apache.mesos.{ Protos => mesos }
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+
+object TaskGroupBuilder {
+  val log = LoggerFactory.getLogger(getClass)
+
+  def build(
+    podDefinition: PodDefinition,
+    offer: mesos.Offer,
+    runningTasks: => Iterable[Task],
+    newInstanceId: PathId => Instance.Id,
+    config: MarathonConf
```

Comment:
OK, I don't think my comment was very clear. I hate MarathonConf. I don't want to see it in new code if we can avoid it. I'd rather see a BuilderConfig case class in the parameter list here vs. MarathonConf.